src/main/java/com/hivemq/persistence/local/memory/ClientSessionMemoryLocalPersistence.java

```diff
@@ -0,0 +1,531 @@
+/*
+ * Copyright 2019 dc-square GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hivemq.persistence.local.memory;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.MetricRegistry;
+import com.google.common.collect.ImmutableSet;
+import com.hivemq.annotations.ExecuteInSingleWriter;
+import com.hivemq.configuration.service.InternalConfigurations;
+import com.hivemq.configuration.service.MqttConfigurationService;
+import com.hivemq.extension.sdk.api.annotations.NotNull;
+import com.hivemq.extension.sdk.api.annotations.Nullable;
+import com.hivemq.logging.EventLog;
+import com.hivemq.metrics.HiveMQMetrics;
+import com.hivemq.persistence.NoSessionException;
+import com.hivemq.persistence.PersistenceEntry;
+import com.hivemq.persistence.PersistenceFilter;
+import com.hivemq.persistence.clientsession.ClientSession;
+import com.hivemq.persistence.clientsession.ClientSessionWill;
+import com.hivemq.persistence.clientsession.PendingWillMessages;
+import com.hivemq.persistence.exception.InvalidSessionExpiryIntervalException;
+import com.hivemq.persistence.local.ClientSessionLocalPersistence;
+import com.hivemq.persistence.local.xodus.BucketChunkResult;
+import com.hivemq.persistence.local.xodus.bucket.BucketUtils;
+import com.hivemq.persistence.payload.PublishPayloadPersistence;
+import com.hivemq.util.ClientSessions;
+import com.hivemq.util.ThreadPreConditions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.hivemq.mqtt.message.connect.Mqtt5CONNECT.SESSION_EXPIRE_ON_DISCONNECT;
+import static com.hivemq.mqtt.message.connect.Mqtt5CONNECT.SESSION_EXPIRY_NOT_SET;
+import static com.hivemq.util.ThreadPreConditions.SINGLE_WRITER_THREAD_PREFIX;
+
+/**
+ * @author Georg Held
+ */
+public class ClientSessionMemoryLocalPersistence implements ClientSessionLocalPersistence {
+
+    private static final @NotNull Logger log = LoggerFactory.getLogger(ClientSessionMemoryLocalPersistence.class);
+
+    final @NotNull AtomicLong currentMemorySize = new AtomicLong();
```

Comment:
can we please order the fields by importance? the memory size should be the last imho